packages/web3-providers/src/entry.ts

```diff
@@ -195,88 +77,63 @@ export { ContractWallet } from './SmartPay/libs/ContractWallet.js'
+export { GoPlusLabs } from './GoPlusLabs/index.js'
 
 // CoinGecko
-export const CoinGeckoTrending = new CoinGeckoTrendingAPI()
-export const CoinGeckoPriceEVM = new CoinGeckoPriceAPI_EVM()
-export const CoinGeckoPriceSolana = new CoinGeckoPriceAPI_Solana()
+export { CoinGeckoTrending } from './CoinGecko/index.js'
 
 // R2D2
 export const R2D2TokenList = new R2D2TokenListAPI()
-export const R2D2Domain = new R2D2DomainAPI()
 
 // Name Service
-export const ENS = new ENS_API()
-export const ARBID = new ARBID_API()
-export const SpaceID = new SpaceID_API()
-export const Bonfida = new BonfidaAPI()
+export { ENS } from './ENS/index.js'
 
 // Debank
-export const DeBankGasOption = new DeBankGasOptionAPI()
-export const DeBankFungibleToken = new DeBankFungibleTokenAPI()
-export const DeBankHistory = new DeBankHistoryAPI()
-
-// Alchemy
-export const AlchemyEVM = new AlchemyEVM_API()
-export const AlchemyFlow = new AlchemyFlowAPI()
+export { DeBankHistory } from './DeBank/index.js'
 
 // NFTScan
-export const NFTScanTrending_EVM = new NFTScanTrendingAPI_EVM()
-export const NFTScanTrending_Solana = new NFTScanTrendingAPI_Solana()
-export const NFTScanNonFungibleTokenEVM = new NFTScanNonFungibleTokenAPI_EVM()
-export const NFTScanNonFungibleTokenSolana = new NFTScanNonFungibleTokenAPI_Solana()
-
-// SID
-export const SID_Domain = new SID_DomainAPI()
+export {
+    NFTScanTrending_EVM,
+    NFTScanTrending_Solana,
+    NFTScanNonFungibleTokenEVM,
+    NFTScanNonFungibleTokenSolana,
+} from './NFTScan/index.js'
 
 // Chainbase
-export const ChainbaseHistory = new ChainbaseHistoryAPI()
-export const ChainbaseFungibleToken = new ChainbaseFungibleTokenAPI()
-export const ChainbaseNonFungibleToken = new ChainbaseNonFungibleTokenAPI()
-export const ChainbaseDomain = new ChainbaseDomainAPI()
-export const ChainbaseRedPacket = new ChainbaseRedPacketAPI()
+export { ChainbaseHistory, ChainbaseDomain } from './Chainbase/index.js'
 
 // Zerion
-export const Zerion = new ZerionAPI()
-export const ZerionNonFungibleToken = new ZerionNonFungibleTokenAPI()
-export const ZerionTrending = new ZerionTrendingAPI()
-export const ZerionGas = new ZerionGasAPI()
+export { Zerion, ZerionGas, ZerionNonFungibleToken, ZerionTrending } from './Zerion/index.js'
 
 // Fuse
-export const Fuse = new FuseAPI()
-export const FuseCoin = new FuseCoinAPI()
-export const FuseNonFungibleCollection = new FuseNonFungibleCollectionAPI()
+export { Fuse, FuseCoin } from './Fuse/index.js'
+export { FuseNonFungibleCollection } from './Fuse/apis/NonFungibleCollection.js'
 
 // Smart Pay
-export const SmartPayBundler = new SmartPayBundlerAPI()
+export { SmartPayBundler } from './SmartPay/index.js'
 export const SmartPayFunder = new SmartPayFunderAPI()
 export const SmartPayOwner = new SmartPayOwnerAPI()
 export const SmartPayAccount = new SmartPayAccountAPI()
 
 // RSS3
-export const RSS3 = new RSS3API()
+export { RSS3 } from './RSS3/index.js'
 
 // Airdrop
 export const Airdrop = new AirdropAPI()
```

Comment:
Export it as a singleton.

Reply:
it creates a new ContractReadonlyAPI inside it, and ContractReadonlyAPI accepts options. I will extract a "default" version of ContractReadonlyAPI so it can be a singleton but not in this PR.